python-avd/pyavd/_eos_designs/schema/schema_fragments/bgp_peer_groups.schema.yml

```diff
@@ -34,6 +34,30 @@ keys:
             documentation_options:
               hide_keys: true
             $ref: "eos_cli_config_gen#/keys/router_bgp/keys/peer_groups/items"
+      use_separate_mlag_peer_group_for_overlay:
+        type: bool
+        default: false
+        description: Configure a different BGP peer group for MLAG for underlay and overlay.
+      mlag_ipv4_overlay_peer:
+        type: dict
+        keys:
+          name:
+            type: str
+            default: MLAG-IPv4-OVERLAY-PEER
```

Comment:
let's remove this default value and in the code:
1/ default it to the value of `mlag_ipv4_underlay_peer.name`
2/ the code will need to check whether `mlag_ipv4_underlay_peer.name` and `mlag_ipv4_vrfs_peer.name` are equal or not.

Create a shared_utils to get the name based on thta
if the names are the same (either because default or being set to the same name as underlay one), then the password, bfd, .. under this key will be ignored. Make a note of this behavior in the description of the key.